### ISSUE TITLE
OptionButton options not selectable after DisplayServer.window_set_size()

### ISSUE THREAD

**Issue by mouring** [godotengine/godot#59573]
### Godot version

4.0 Alpha 5

### System information

MacOS 12.3 on Intel MacBook Pro 

### Issue description

After issuing a DisplayServer.window_set_size(..) the OptionButton no longer accepts selection of options via mouse, but can be selected via keyboard.

This worked correctly using the OS.set_window_size() in Godot 3.4.4.

### Steps to reproduce

* Create a scene (attached project uses 2D Scene, but my main code uses a Control node)
* Add in an "OptionButton" to the scene
* Attach script to button 
* Add the following to the script:

```
func _ready():
	add_item("Item 1", 0)
	add_item("Item 2", 1)
	add_item("Item 3", 2)

	DisplayServer.window_set_size(Vector2(1024*2, 600*2))
```

### Minimal reproduction project

[Uploading OptionButtonTest.zip…]()

**Comment by Sauermann** [comment]
Please update the link to your Minimal reproduction project. I am unable to download it.

**Comment by Zireael07** [comment]
It failed uploading - they'll need to reupload it again.

**Comment by mouring** [comment]
[OptionButtonTest.zip](https://github.com/godotengine/godot/files/8357808/OptionButtonTest.zip)

Re uploading test, not sure why it failed the first time.

**Comment by Sauermann** [comment]
I can confirm this bug on Linux.
This gets fixed by #59118 or #59310.

**Comment by mouring** [comment]
Thank you.  I assumed this was covered somewhere, but it wasn't clear what search term to use.
